src/base/searchengine.cpp

```diff
@@ -104,19 +104,26 @@ QStringList SearchEngine::enabledPlugins() const
     return plugins;
 }
 
-QStringList SearchEngine::supportedCategories() const
+QStringList SearchEngine::supportedCategories(const QString &name) const
 {
-    QStringList result;
-    foreach (const PluginInfo *plugin, m_plugins.values()) {
-        if (plugin->enabled) {
-            foreach (QString cat, plugin->supportedCategories) {
-                if (!result.contains(cat))
-                    result << cat;
-            }
-        }
+    QStringList plugins;
+    if (name == "all")
+        plugins = allPlugins();
+    else if ((name == "enabled") || (name == "multi"))
+        plugins = enabledPlugins();
+    else
+        plugins << name.trimmed();
+
+    QSet<QString> categories;
+    for (const QString &name : plugins) {
+        const PluginInfo *plugin = pluginInfo(name);
+        if (!plugin) continue; // plugin wasn't found
+        for (const QString &category : plugin->supportedCategories)
+            if (!categories.contains(category))
```

Comment:
by using `QSet`, this `if()` is redundant now.

Reply:
Yes. Otherwise, why this change was suggested?..